WordPress/Classes/ViewRelated/NUX/SigninAccountAndBlogViewController.swift

```diff
@@ -0,0 +1,656 @@
+                                                        self.usernameField.text = username
+                                                    }
+                                                    if let password = loginDict[WPOnePasswordPasswordKey]  as? String {
+                                                        self.loginFields.password = password
+                                                        self.usernameField.text = password
+                                                    }
+
+                                                    WPAnalytics.track(.OnePasswordSignup)
+
+                                                    // Note: Since the Site field is right below the 1Password field, let's continue with the edition flow
+                                                    // and make the SiteAddress Field the first responder.
+                                                    self.siteURLField.becomeFirstResponder()
+        }
+
+    }
+
+
+    @IBAction func handleTermsOfServiceButtonTapped(sender: UIButton) {
+        let url = NSURL(string: WPAutomatticTermsOfServiceURL)
+        let controller = WPWebViewController(URL: url)
+        let navController = RotationAwareNavigationViewController(rootViewController: controller)
+        presentViewController(navController, animated: true, completion: nil)
+    }
+
+
+    // MARK: - Keyboard Notifications
+    
+    
+    func handleKeyboardWillShow(notification: NSNotification) {
+        keyboardWillShow(notification)
+    }
+    
+    
+    func handleKeyboardWillHide(notification: NSNotification) {
+        keyboardWillHide(notification)
+    }
+}
+
+
+extension SigninAccountAndBlogViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(textField: UITextField) -> Bool {
+        if textField == emailField {
+            usernameField.becomeFirstResponder()
+        } else if textField == usernameField {
+            passwordField.becomeFirstResponder()
+        } else if textField == passwordField {
+            siteURLField.becomeFirstResponder()
+        } else if submitButton.enabled {
+            validateForm()
+        }
+        return true
+    }
+
+
+    func textFieldShouldEndEditing(textField: UITextField) -> Bool {
+        if textField != emailField || didCorrectEmailOnce {
+            return true
+        }
+
+        let suggestedEmail = EmailCheckerHelper.suggestDomainCorrection(textField.text)
+        if suggestedEmail != textField.text {
+            textField.text = suggestedEmail
+            didCorrectEmailOnce = true
+            loginFields.emailAddress = textField.nonNilTrimmedText()
+        }
+        return true
+    }
+
+
+    func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
+        // Disallow spaces except for the password field
+        if textField != passwordField && string == " " {
```

Comment:
I don't think this form will change soon, but when it does, this seems like a good spot to hide a bug. I'd rather be explicit about which fields disallow spaces rather than "anything but the password"